service/access.go

```diff
@@ -307,15 +319,25 @@ func (a *accessService) updateAccessToken(ctx context.Context, domain, role, pro
 			return nil, e
 		}
 
+		tok, _, err := jwt.NewParser().ParseUnverified(at.AccessToken, &jwt.RegisteredClaims{})
+		if err != nil {
+			return nil, fmt.Errorf("jwt.ParseUnverified() err: %v", err)
+		}
+
+		expTime, err := tok.Claims.GetExpirationTime()
+		if err != nil {
+			return nil, fmt.Errorf("jwt.GetExpirationTime() err: %v", err)
+		}
+
 		a.tokenCache.SetWithExpire(key, &accessCacheData{
-			token:             at,
+			token:             at.AccessToken,
 			domain:            domain,
 			role:              role,
 			proxyForPrincipal: proxyForPrincipal,
-			expiresIn:         expiresIn,
-		}, time.Unix(at.ExpiresIn, 0).Sub(expTimeDelta))
+			expiry:            expTime.Unix(),
```

Comment:
```suggestion
			expiresIn:            expiresIn,
			expiry:            expTime.Unix(),
```
cache request parameter

Reply:
corrected in https://github.com/AthenZ/athenz-client-sidecar/pull/13/commits/636bfc04a8cfa4823f2aa6555787b701b1db21c1